src/kaltura-player.js

```diff
@@ -535,6 +587,82 @@ class KalturaPlayer extends FakeEventTarget {
     return this._localPlayer.Error;
   }
 
+  _onChangeSourceStarted(): void {
+    this._configureOrLoadPlugins(this._pluginsConfig);
```

Comment:
why don't you use it in setMedia instead?

Reply:
I just mimic the [previous implement ](https://github.com/kaltura/playkit-js/pull/463/files#diff-29cfd888e32c875acb1a2dd113e516dbL486) in playkit

Reply:
I see but I think it'll be cleaner to put it on `setMedia` instead if there isn't another reason

Reply:
sources could be passed via `configure` on the config object. we can't rely on `setMedia` only

Reply:
so why not use setMedia always, also internally?
basically this is can almost be considered a bug as we have logic around adding sources to player which is enforced in setmedia.
the suggestion here is to also use setMedia if the sources are provided in the config passed to setup call.

Reply:
agreed but I said above this is out of scope of this change so I prefer to do it separately